### PR TITLE
Fix typos in JS AddHostObjectAsyncProxyBase & AddHostObjectSyncProxy

### PR DESCRIPTION
**Pull request by srwei**
Rendered Ref topics for review:
* **HostObjectAsyncProxyBase class**
   * https://review.learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase?branch=pr-en-us-3243
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase)
* **HostObjectSyncProxy class**
   * https://review.learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy?branch=pr-en-us-3243
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy)

GitHub Issue:
https://github.com/MicrosoftDocs/edge-developer/issues/2751

AB#53042077

### CODE REVIEW

File: microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml [hunk 1]
```diff
@@ -125,14 +125,13 @@ methods:
...
         - id: propertyValue
           description: Value to set the property to.
           type: any
       return:
         type: any
         description: >-
-          A promise representing the value of the property after it is set. This promise only resolves after the
-          property changes value.
+          The value of the property after it is set.
 extends: <a href="https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.callablefunction.html">CallableFunction</a>
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml:137]: 404 #1
i confirmed that in the live docs (& this PR), the link "Extends CallableFunction" is 404.
in live page:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase
[CallableFunction] url is:
https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.callablefunction.html
that's 404
line 138 in main branch:
https://github.com/srwei/edge-developer/blob/main/microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml
= 137 in pr's branch

File: microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml [hunk 2]
```diff
@@ -135,7 +135,7 @@ methods:
       content: 'setLocalProperty(propertyName: string, propertyValue: any): any;'
       parameters:
         - id: propertyName
-          description: Name of the property to get the value of.
+          description: Name of the property of which to set the value.
           type: string
         - id: propertyValue
           description: Value to set the property to.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml:141]: line 146: The link "Extends CallableFunction" is 404.
That link in live docs is in:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy

**Review by mikehoffms — CHANGES_REQUESTED** [review]
Fix links per build status report / 2 PR Comments.  per Writer/Editor review.

File: microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml [hunk 3]
```diff
@@ -134,4 +134,4 @@ methods:
         type: any
         description: >-
           The value of the property after it is set.
-extends: <a href="https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.callablefunction.html">CallableFunction</a>
+extends: <a href="https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase">CallableFunction</a>
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml:137]:
```suggestion
extends: <a href="https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase">CallableFunction</a>
```
omit locale per corp style guide

File: microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml [hunk 4]
```diff
@@ -135,12 +135,12 @@ methods:
...
+          description: Name of the property of which to set the value.
           type: string
         - id: propertyValue
           description: Value to set the property to.
           type: any
       return:
         type: any
         description: The value of the property after it is set.
-extends: <a href="https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.callablefunction.html">CallableFunction</a>
+extends: <a href="https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy">CallableFunction</a>
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml:146]:
```suggestion
extends: <a href="https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy">CallableFunction</a>
```

Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml:146]: in Markdown tagging, LMC links would omit prefix:
/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy
not sure if applies - build report will indicate

Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml:146]: you might be still troubleshooting link.  The link text is "**CallableFunction**", but the new link destination 
https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy
has a contradictory title: "**HostObjectSyncProxy class**"

Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml]: closing Outdated comment.  Dave is researching where CallableFunciton is defined.

File: microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml]: en-us/ removed.  Closing comment.  Dave is researching where CallableFunciton is defined.

File: microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml (hunk 1 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml]: en-us/ removed.  Closing comment.  Dave is researching where CallableFunciton is defined.

File: microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml:141]: Closing comment.  Dave is researching where CallableFunciton is defined.

File: microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml [hunk 5]
```diff
@@ -135,12 +135,12 @@ methods:
...
+          description: Name of the property of which to set the value.
           type: string
         - id: propertyValue
           description: Value to set the property to.
           type: any
       return:
         type: any
         description: The value of the property after it is set.
-extends: <a href="https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.callablefunction.html">CallableFunction</a>
+extends: <a href="https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy">CallableFunction</a>
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/hostobjectsyncproxy.yml:146]:
```suggestion
extends: <a href="https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Function">Function</a>
```
@david-risney @srwei

File: microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml [hunk 6]
```diff
@@ -125,14 +125,13 @@ methods:
...
           description: Value to set the property to.
           type: any
       return:
         type: any
         description: >-
-          A promise representing the value of the property after it is set. This promise only resolves after the
-          property changes value.
-extends: <a href="https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.callablefunction.html">CallableFunction</a>
+          The value of the property after it is set.
+extends: <a href="https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase">CallableFunction</a>
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml:137]:
```suggestion
extends: <a href="https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Function">Function</a>
```
@srwei @david-risney

**Review by mikehoffms — APPROVED** [review]
Approved PR per Writer/Editor review

**Review by david-risney — APPROVED** [review]

**Pull request merged**